grails-app/views/product/_mergeProducts.gsp

```diff
@@ -0,0 +1,82 @@
+<div class="dialog">
+    <p class="message m-2">
+        <warehouse:message code="product.mergeProducts.description.label"/>
+    </p>
+    <table>
+        <tbody>
+            <tr class="prop">
+                <td valign="top" class="name">
+                    <label for="primary-product">
+                        <warehouse:message code="product.mergeProducts.primaryProduct.label"/>
+                    </label>
+                </td>
+                <td valign="top" class="value">
+                    <g:autoSuggest
+                       id="primary-product"
+                       name="primary-product"
+                       jsonUrl="${request.contextPath}/json/findProductByName?skipQuantity=true"
+                       styleClass="text"
+                       showColor="true"
+                        valueId="${primaryProduct.id}"
+                        valueName="${primaryProduct.productCode} ${primaryProduct.name}"
+                    />
+                </td>
+            </tr>
+            <tr class="prop">
+                <td valign="top" class="name">
+                    <label for="obsolete-product">
+                        <warehouse:message code="product.mergeProducts.obsoleteProduct.label"/>
+                    </label>
+                </td>
+                <td valign="top" class="value">
+                    <g:autoSuggest
+                        id="obsolete-product"
+                        name="obsolete-product"
+                        jsonUrl="${request.contextPath}/json/findProductByName?skipQuantity=true"
+                        styleClass="text"
+                        showColor="true"
+                    />
+                </td>
+            </tr>
+        </tbody>
+        <tfoot>
+            <tr class="prop">
+                <td></td>
+                <td valign="top" class="value">
+                    <button
+                        id="merge-button"
+                        type="button"
+                        class="button icon approve"
+                    >
+                        <g:message code="product.mergeProducts.label" default="Merge Products"/>
+                    </button>
+                </td>
+            </tr>
+        </tfoot>
+    </table>
+    <g:render template="/product/mergeProductConfirmationDialog" />
+</div>
+<script type="text/javascript">
+  $(function() {
+   $("#merge-button")
+     .click(function () {
+       const primaryProductId = $("#primary-product-id").val();
+       const obsoleteProductId = $("#obsolete-product-id").val();
+
+       if (primaryProductId && obsoleteProductId) {
+         const warningMessage = $("#merge-product-warning-message")
+           .text()
+           .replaceAll("{0}", primaryProductId)
+           .replaceAll("{1}", obsoleteProductId);
+         $("#merge-product-warning-message").text(warningMessage)
```

Comment:
@awalkowiak here is how I resolved my problem with inserting product id's into translatable messages.
Basically I added {0} and {1} as arguments in translatable message.properties label and when rendering this label I don't pass argument into  `${warehouse.message(code: 'product.mergeProducts.confirmation.warning.label')}` and then I manually (using jquery) **replaceAll** strings that are **{0}** and **{1}**  with appropriate product id's

Reply:
I guess it is ok considering when it is rendering. I can't see a better solution for now.